src/generic_ui/polymer/contact.ts

```diff
@@ -111,10 +115,74 @@ Polymer({
       },
       userId: this.contact.userId // Cloud instance userId
     }).then((cloudInviteUrl: string) => {
-      ui.showDialog(ui.i18n_t("CLOUD_SHARE_INSTRUCTIONS"), '', ui.i18n_t("OK"),
+      this.ui.showDialog(this.ui.i18n_t("CLOUD_SHARE_INSTRUCTIONS"), '', this.ui.i18n_t("OK"),
         undefined, cloudInviteUrl);
     });
   },
+  displayCloudRemovalConfirmation: function() {
+    // We keep track of which contact should be deleted with toBeDeleted
+    // TODO: Instead of using toBeDeleted, send the contact's ID as data
+    // with a core-signal
+    this.toBeDeleted = true;
+    // Core signals for the confirmation dialog cancel and continue buttons
+    var coreSignals = {
+      'cancel': 'cancel-cloud-removal',
+      'continue': 'remove-cloud-server'
+    };
+    if (this.contact.status === this.UserStatus.CLOUD_INSTANCE_CREATED_BY_LOCAL) {
+      this.ui.getConfirmation(this.ui.i18n_t("REMOVE_CLOUD_SERVER"), this.ui.i18n_t('DESTROY_CLOUD_SERVER_CONFIRMATION'),
```

Comment:
I'm not sure you need to do all this stuff with the ```coreSignals``` and ```toBeDeleted```.  getConfirmation returns a Promise which fulfills if the user confirms or rejects if they cancel.  Is it possible to just use that returned Promise, so you could probably write something like:
```this.ui.getConfirmation(..., ..., true).then(() => {
  // code from removeCloudFriend
});
```
```